algorithms/merging/merge.py

```diff
@@ -114,76 +131,103 @@ def make_MAD_merged_mtz_file(params, experiments, reflections, wavelengths):
 
         logger.info("Running merge for wavelength: %s", wavelength)
         merged, anom, amplitudes, anom_amp = merge_and_truncate(
-            params, new_exps, [refls]
+            new_exps,
+            [refls],
+            d_min=d_min,
+            d_max=d_max,
+            combine_partials=combine_partials,
+            partiality_threshold=partiality_threshold,
+            anomalous=anomalous,
+            assess_space_group=assess_space_group,
+            truncate=truncate,
+            n_residues=n_residues,
+            n_bins=n_bins,
+            use_internal_variance=use_internal_variance,
+            show_wilson_stats=show_wilson_stats,
+            show_merging_stats=show_merging_stats,
         )
         #### Add each wavelength as a new crystal.
-        mtz_writer.add_crystal(
-            crystal_name=cname, project_name=params.output.project_name
-        )
+        mtz_writer.add_crystal(crystal_name=cname, project_name=project_name)
         mtz_writer.add_empty_dataset(wavelength, name=dname)
         mtz_writer.add_dataset(merged, anom, amplitudes, anom_amp)
 
     return mtz_writer.mtz_file
 
 
 def make_merged_mtz_file(
-    params,
     wavelength,
     merged_array,
     merged_anomalous_array=None,
     amplitudes=None,
     anomalous_amplitudes=None,
+    crystal_name=None,
+    dataset_name=None,
+    project_name=None,
 ):
     """Make an mtz object for the data, adding the date, time and program."""
 
     assert merged_array.is_xray_intensity_array()
 
     mtz_writer = MergedMTZWriter(merged_array.space_group(), merged_array.unit_cell())
     mtz_writer.add_crystal(
-        crystal_name=params.output.crystal_names[0],
-        project_name=params.output.project_name,
+        crystal_name=crystal_name, project_name=project_name,
     )
-    mtz_writer.add_empty_dataset(wavelength, name=params.output.dataset_names[0])
+    mtz_writer.add_empty_dataset(wavelength, name=dataset_name)
     mtz_writer.add_dataset(
         merged_array, merged_anomalous_array, amplitudes, anomalous_amplitudes
     )
 
     return mtz_writer.mtz_file
 
 
-def merge_and_truncate(params, experiments, reflections):
+def merge_and_truncate(
+    experiments,
+    reflections,
+    d_min=None,
+    d_max=None,
+    combine_partials=True,
+    partiality_threshold=0.4,
+    anomalous=True,
+    use_internal_variance=False,
+    assess_space_group=False,
+    truncate=True,
+    n_residues=200,
+    n_bins=20,
+    show_wilson_stats=False,
```

Comment:
Looking over the parameter list, I don't think we need the two show_* parameters. This information should always be logged, or if it is not very useful then perhaps demoted to debug level. Then log output can be suppressed in the usual way if not needed when being called by other code. If you agree I'll make the change.